### PR TITLE
Push financial data into apply api on submission of application

### PR DESCRIPTION
**Pull request by chrisfoster76**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 1]
```diff
@@ -487,6 +499,93 @@ FROM OPENJSON(i.Sequences)
...
+                                    return item;
+                                }
+                            ), parameters, null, true, "ApplicationId");
+
+                    return results.ToList();
+                }
+            }
+            catch (Exception ex)
+            {
+                var m = ex.Message;
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:584]: Were you expecting to log this or use it for debug?

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 2]
```diff
@@ -487,6 +499,93 @@ FROM OPENJSON(i.Sequences)
...
+                financialStatusNew = FinancialReviewStatus.New,
+                financialStatusInProgress = FinancialReviewStatus.InProgress,
+                gatewayStatusPass = GatewayReviewStatus.Pass
+            };
+
+            try
+            {
+
+
+
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:566]: bit of whitespace here

**Review by martin-hornsby — DISMISSED** [review]

**Review by martin-hornsby — DISMISSED** [review]

**Review by martin-hornsby — DISMISSED** [review]

File: src/SFA.DAS.ApplyService.Application/Apply/Roatp/RoatpPreambleQuestionBuilder.cs [hunk 3]
```diff
@@ -368,6 +373,15 @@ public static class RoatpWorkflowQuestionTags
...
         public static string UKRLPVerificationCharityRegNumber = "UKRLPVerificationCharityRegNumber";
+        public static string Turnover = "Turnover";
+        public static string Depreciation = "Depreciation";
+        public static string ProfitLoss = "ProfitLoss";
+        public static string Dividends = "Dividends";
+        public static string IntangibleAssets = "IntangibleAssets";
+        public static string Assets = "Assets";
+        public static string Liabilities = "Liabilities";
+        public static string ShareholderFunds = "ShareholderFunds";
+        public static string Borrowings = "Borrowings";
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Application/Apply/Roatp/RoatpPreambleQuestionBuilder.cs:384]: As mentioned. Best to prefix these in QnA so they have context in the wider context of things

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs [hunk 4]
```diff
@@ -509,6 +521,80 @@ FROM OPENJSON(i.Sequences)
...
+                            apply.FinancialGrade AS FinancialReviewDetails,
+                            org.Name AS OrganisationName,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.UKPRN') AS Ukprn,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ReferenceNumber') AS ApplicationReferenceNumber,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ProviderRouteName') AS ApplicationRoute,
+                            JSON_VALUE(apply.ApplyData, '$.ApplyDetails.ApplicationSubmittedOn') AS SubmittedDate,
+                            JSON_VALUE(apply.ApplyData, '$.GatewayReviewDetails.OutcomeDateTime') AS GatewayOutcomeDateTime,
+                            JSON_VALUE(apply.ApplyData, '$.GatewayReviewDetails.CompaniesHouseDetails.CompanyNumber') AS CompanyNumber,
+                            JSON_VALUE(apply.ApplyData, '$.GatewayReviewDetails.CharityCommissionDetails.CharityNumber') AS CharityNumber,
+                            CASE s.NotRequired WHEN 'false' THEN 'Not exempt' ELSE 'Exempt' END AS DeclaredInApplication,
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:542]: Isn't this a bit field in the database? Would have expected 0

**Review by martin-hornsby — CHANGES_REQUESTED** [review]

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.ApplyService.Data/ApplyRepository.cs (hunk 4 above)
Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.Data/ApplyRepository.cs:542]: please ignore, it's doing a transform in the Cross Apply

**Review by martin-hornsby — APPROVED** [review]

**Review by martin-hornsby — APPROVED** [review]

**Pull request merged**